packages/data-provider/src/services/submission/processor.ts

```diff
@@ -421,6 +423,39 @@ const processor = (dependencies: BaseDependencies) => {
 			dataValidated: dataMergedByEntityName,
 		});
 
+		// Check for records to be updated that its systemId was not found in the Submitted Data collection.
+		// Any error found will cause the submission to be marked as 'invalid'
+		Object.entries(submissionData.updates ?? {}).forEach(([entityName, recordsToUpdate]) => {
+			recordsToUpdate.forEach((submissionEditData, index) => {
+				const found = findEditSubmittedData(entityName, submissionEditData.systemId, dataMergedByEntityName);
+
+				if (found) {
+					return;
+				}
+
+				logger.error(
+					LOG_MODULE,
+					`Record with systemId '${submissionEditData.systemId}' not found in entity '${entityName}'`,
+				);
+
+				if (!submissionSchemaErrors.updates) {
+					submissionSchemaErrors.updates = {};
+				}
+
+				if (!submissionSchemaErrors.updates[entityName]) {
+					submissionSchemaErrors.updates[entityName] = [];
+				}
+
+				submissionSchemaErrors.updates[entityName].push(
+					createUnrecognizedFieldBatchError({
```

Comment:
Is this the correct error name? The field `systemId` is recognized, but the value is not found for this entity. This is an invalid value, not unrecognized field.